build.gradle

```diff
@@ -390,6 +390,7 @@ integTest {
     systemProperty("https", is_https)
     systemProperty("user", user)
     systemProperty("password", password)
+    systemProperty("test.exhaustive", System.getProperty("test.exhaustive"))
```

Comment:
Added this flag in order to provide options for more exhaustive testing. In this PR, I put the index size checks wrapped in it to avoid flakiness. @navneet1v

Reply:
make sense. I think this is a good idea for now.